src/gameplay/tests/sinad.test.ts

```diff
@@ -0,0 +1,192 @@
+import  { GameConfig, isSubsetOf }  from '../game';
+import Game from '../game';
+import { RoomConfig } from '../room';
+import { GameMode } from '../gameModes';
+import { ReadyPrompt } from '../../sockets/readyPrompt';
+import { RoomCreationType } from '../roomTypes';
+
+
+jest.mock('../gameWrapper');
+
+
+const vh = {
+  'Piplup':
+    [
+      [
+        'VHpicked VHreject',
+        'VHapprove',
+      ],
+      [
+        'VHapprove',
+      ],
+      [
+        'VHleader VHapprove',
+      ],
+    ],
+  'Torchic':
+    [
+      [
+        'VHpicked VHleader VHreject',
+        'VHapprove',
+      ],
+      [
+        'VHapprove',
+      ],
+      [
+        'VHapprove',
+      ],
+    ],
+  'Mudkip':
+    [
+      [
+        'VHreject',
+        'VHpicked VHapprove',
+      ],
+      [
+        'VHpicked VHapprove',
+      ],
+      [
+        'VHpicked VHapprove',
+      ],
+    ],
+  'Turtwig':
+    [
+      [
+        'VHreject',
+        'VHleader VHapprove',
+      ],
+      [
+        'VHapprove',
+      ],
+      [
+        'VHpicked VHapprove',
+      ],
+    ],
+  'MewTwo':
+    [
+      [
+        'VHreject',
+        'VHapprove',
+      ],
+      [
+        'VHpicked VHleader VHapprove',
+      ],
+      [
+        'VHpicked VHapprove',
+      ],
+    ],
+  'Charmander':
+    [
+      [
+        'VHreject',
+        'VHpicked VHapprove',
+      ],
+      [
+        'VHpicked VHapprove',
+      ],
+      [
+        'VHpicked VHapprove',
+      ],
+    ],
+};
+
+const roomConfig: RoomConfig = new RoomConfig(
+  '1',
+  1,
+  jest.fn(),
+  10,
+  '', //password
+  GameMode.AVALON,
+  false,
+  new ReadyPrompt(),
+);
+const gameConfig: GameConfig = new GameConfig(
+  roomConfig,
+  false,
+  false,
+  false,
+  RoomCreationType.CUSTOM_ROOM,
+  jest.fn(),
+);
+
+const mockGame: Game = new Game(gameConfig);
+
+
+const adjustGame = (game: Game) => {
+ 
+      game.voteHistory = vh; 
+      game.enableSinadMode = true;
+      game.playersInGame = ['Piplup', 'Torchic', 'Mudkip', 'Turtwig', 'MewTwo', 'Charmander'];
+      game.gameMode = GameMode.AVALON;
+      game.missionHistory = ['succeeded','succeeded','failed'];
+};
+
+
+describe('Sinad mode', () => {
+  beforeEach( () => {
+    adjustGame(mockGame);
+  });
+
+
+  it('should only work for 6p', () => {
+      expect(mockGame.shouldSinadRun()).toEqual(true);
+      mockGame.playersInGame.push('Squirtle');
+      expect(mockGame.shouldSinadRun()).toEqual(false);
+  });
+
+  it('should not run if m1 failed', () => {
```

Comment:
Can you move these tests to the existing game.test.ts? 

That way you don't have to use mocks and you can get better test coverage.